lib/oxidized/model/ios.rb

```diff
@@ -26,15 +26,21 @@ class IOS < Oxidized::Model
   cmd :secret do |cfg|
     cfg.gsub! /^(snmp-server community).*/, '\\1 <configuration removed>'
     cfg.gsub! /^(snmp-server host \S+( vrf \S+)?( version (1|2c|3))?)\s+\S+((\s+\S*)*)\s*/, '\\1 <secret hidden> \\5'
-    cfg.gsub! /^(username \S+ privilege \d+) (\S+).*/, '\\1 <secret hidden>'
-    cfg.gsub! /^(username \S+ password \d) (\S+)/, '\\1 <secret hidden>'
-    cfg.gsub! /^(username \S+ secret \d) (\S+)/, '\\1 <secret hidden>'
-    cfg.gsub! /^(enable (password|secret) \d) (\S+)/, '\\1 <secret hidden>'
+    cfg.gsub! /^(username .+ (password|secret) \d) .+/, '\\1 <secret hidden>'
```

Comment:
The "username" command may include additional options. Previously, with their presence secrets were not cut.
before:
```
username TEST2 access-class 2 privilege 0 noescape secret 5 $1$qDrb$WZiqdvbykLL.x0fj59fT10
```

after:
```
username TEST2 access-class 2 privilege 0 noescape secret 5 <secret hidden>
```

Reply:
Why was the match for `enable` also removed here? That case does not appear to be handled by the suggested replacement regexp.

Reply:
The match for ``enable`` is not deleted, but changed. Disable the display of comments on the code, then you will see the change completely, it includes:
```
- cfg.gsub! /^(enable (password|secret) \d) (\S+)/, '\\1 <secret hidden>'
+ cfg.gsub! /^(enable (password|secret)( level \d+)? \d) .+/, '\\1 <secret hidden>'
```
before:
```
enable secret level 3 5 $1$iCV/$xgCDHtF4OhgjnFhd/ff1G.
enable secret 5 <secret hidden>
```
after:
```
enable secret level 3 5 <secret hidden>
enable secret 5 <secret hidden>
```

Commenting on lines in patch code was a bad idea. It was necessary to describe all the changes in the pull-request body.

Reply:
The mistake is mine, glanced over this too quickly instead of looking at all the change set. The comments have in fact been helpful!